examples/fabric-admin/main.cpp

```diff
@@ -23,11 +23,34 @@
 #include "commands/pairing/Commands.h"
 #include <zap-generated/cluster/Commands.h>
 
+#include <iostream>
+#include <string>
+#include <vector>
+
 // ================================================================================
 // Main Code
 // ================================================================================
 int main(int argc, char * argv[])
 {
+    // Convert command line arguments to a vector of strings for easier manipulation
+    std::vector<std::string> args(argv, argv + argc);
+
+    // Check if "interactive" and "start" are not in the arguments
+    if (args.size() < 3 || args[1] != "interactive" || args[2] != "start")
+    {
+        // Insert "interactive" and "start" after the executable name
+        args.insert(args.begin() + 1, "interactive");
+        args.insert(args.begin() + 2, "start");
+
+        // Update argc and argv to reflect the new arguments
+        argc = static_cast<int>(args.size());
+        for (size_t i = 0; i < args.size(); ++i)
+        {
+            argv[i] = const_cast<char *>(args[i].c_str());
```

Comment:
You cannot assign to `argv` as you are increasing its size.

Simply allocate make a std::vector<const char*> and pass `.data()` of that vector to commands.Run().

In fact, `args` could be converted to a std::vector<const char*>, filled with the `argv[i]` values by default.

Reply:
Acked